### PR TITLE
add more ice unit test. Refactored some ice code. Fix edge case in po…

### PR DESCRIPTION
**Pull request by chehefen**
…pulateIpFromString

*Issue #, if available:*

*Description of changes:*
- added more ice unit test.
- renamed findCandidateWithConnectionHandle to findCandidateWithSocketConnection because connection has been replaced with SocketConnection
- fix edge case in populateIpFromString
- refactoring

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/Ice/IceAgent.c [hunk 1]
```diff
@@ -287,7 +287,7 @@ STATUS iceAgentReportNewLocalCandidate(PIceAgent pIceAgent, PIceCandidate pIceCa
     return retStatus;
 }
 
-STATUS iceAgentAddRemoteCandidate(PIceAgent pIceAgent, PCHAR pIceCandidateString)
+STATUS  iceAgentAddRemoteCandidate(PIceAgent pIceAgent, PCHAR pIceCandidateString)
```
Comment (zhiyua-git) [inline src/source/Ice/IceAgent.c]: Remove extra space

Comment (MushMal) [inline src/source/Ice/IceAgent.c]: revert

File: src/source/Ice/IceUtils.c [hunk 2]
```diff
@@ -183,7 +183,7 @@ STATUS populateIpFromString(PKvsIpAddress pKvsIpAddress, PCHAR pBuff)
 
     curr = pBuff;
     tail = pBuff + STRLEN(pBuff);
-    while ((next = STRNCHR(curr, tail - curr, '.')) != NULL) {
+    while ((next = STRNCHR(curr, tail - curr, '.')) != NULL && octet < 3) {
```
Comment (zhiyua-git) [inline src/source/Ice/IceUtils.c]: Why octet < 3? Put a constant or comment.

Comment (chehefen) [reply src/source/Ice/IceUtils.c]: added new comment "first 3 octet should always end with a '.', the last octet may end with ' ' or '\0""

File: src/source/Ice/IceAgent.c (hunk 1 above)
Comment (chehefen) [reply src/source/Ice/IceAgent.c]: done
